src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchWriterUtils.cs

```diff
@@ -172,14 +212,20 @@ internal static void WriteChangeSetPreamble(TextWriter writer, string changeSetB
         /// <param name="writer">Writer to write headers.</param>
         /// <param name="inChangeSetBound">Whether we are in ChangeSetBound.</param>
         /// <param name="contentId">The Content-ID value to write in ChangeSet head.</param>
-        private static void WriteHeaders(TextWriter writer, bool inChangeSetBound, string contentId)
+        /// <param name="dependsOnIds">The DependsOn-IDs value to be written if it is not empty.</param>
+        private static void WriteHeaders(TextWriter writer, bool inChangeSetBound, string contentId, string dependsOnIds)
         {
             writer.WriteLine("{0}: {1}", ODataConstants.ContentTypeHeader, MimeConstants.MimeApplicationHttp);
             writer.WriteLine("{0}: {1}", ODataConstants.ContentTransferEncoding, ODataConstants.BatchContentTransferEncoding);
             if (inChangeSetBound && contentId != null)
             {
                 writer.WriteLine("{0}: {1}", ODataConstants.ContentIdHeader, contentId);
             }
+
+            if (!string.IsNullOrEmpty(dependsOnIds))
+            {
+                writer.WriteLine("{0}: {1}", ODataConstants.DependsOnIdsHeader, dependsOnIds);
+            }

```

Comment:
Sorry; I think I wasn't clear on my comment on DependencyIds in multipart/mixed.  I wasn't suggesting that we find a way to write out/read in dependencyIds in multipart/mixed -- that would be interpreting the existing format beyond what the protocol currently defines. The protocol currently says that, outside of a changeset, every statement is implicitly dependent upon the previous one.  So we shouldn't be writing out, or reading in, anything explicit for dependencyIds. Instead, when given a set of dependencyIds we can validate that all specified Ids are valid Ids of previous statements, and in populating dependencyids in multipartmixed we can include a list of all previously read statement ids (which would then make validation work).

Reply:
rolled back the previous implementation. Please see the other comment of this similar issue, about the dependsOn Ids for top-level requests.

---
In reply to: [156802615](https://github.com/OData/odata.net/pull/992#discussion_r156802615) [](ancestors = 156802615)